src/components/WheelPicker/usePresenter.ts

```diff
@@ -1,59 +1,57 @@
 import _ from 'lodash';
 import React from 'react';
-import {ItemProps} from './Item';
+import {LogService} from '../../services';
 import useMiddleIndex from './helpers/useListMiddleIndex';
+import {ItemProps} from './Item';
 
-export type ItemValueTypes = ItemProps | number | string;
+//TODO: deprecate this type
+export type ItemValueTypes = /* ItemProps |  */number | string;
 
-type PropTypes = {
-  initialValue?: ItemValueTypes;
+type PropTypes<T> = {
+  initialValue?: T;
   children?: JSX.Element | JSX.Element[];
-  items?: ItemProps[];
+  items?: ItemProps<T>[];
   itemHeight: number;
   preferredNumVisibleRows: number;
 };
 
-type RowItem = {
-  value: string | number;
+type RowItem<T> = {
+  value: T;
   index: number;
 };
 
-interface Presenter {
-  items: ItemProps[];
-  index: number;
-  height: number;
-  getRowItemAtOffset: (offset: number) => RowItem;
-}
-
-const usePresenter = ({
-  initialValue = 0,
+const usePresenter = <T extends string | number = number>({
+  initialValue,
   children,
   items: propItems,
   itemHeight,
   preferredNumVisibleRows
-}: PropTypes): Presenter => {
-  const extractItemsFromChildren = (): ItemProps[] => {
+}: PropTypes<T>) => {
+
+  const extractItemsFromChildren = (): ItemProps<T>[] => {
     const items = React.Children.map(children, child => {
-      const childAsType: ItemProps = {value: child?.props.value, label: child?.props.label};
+      const childAsType: ItemProps<T> = {value: child?.props.value, label: child?.props.label};
       return childAsType;
     });
     return items || [];
   };
-
   const items = children ? extractItemsFromChildren() : propItems || [];
   const middleIndex = useMiddleIndex({itemHeight, listSize: items.length});
 
   const getSelectedValueIndex = () => {
-    if (_.isString(initialValue) || _.isNumber(initialValue)) {
-      return _.findIndex(items, {value: initialValue});
+    if (_.isObject(initialValue)) {
+      LogService.warn('UILib WheelPicker will stop supporting initialValue prop type as an object (ItemProps). Please pass string or number only');
```

Comment:
I'm assuming this is what this refers to: `export type ItemValueTypes = /* ItemProps |  */number | string;`
If we're only starting this migration we should probably only remove the type in the next major when we remove it.

Reply:
We still support it so functionality won't break but they will get TS error that should force them to migrate or add ignore. WDYT?

Reply:
We don't usually do that IMO, we don't cause TS errors if we can avoid it

Reply:
I remane it WheelPickerItemValue and defined it in types so I can export it from WheelPicker. I also exported outside